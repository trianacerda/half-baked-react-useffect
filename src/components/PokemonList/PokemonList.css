.pokemonList {
  display: flex;
  list-style: none;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  margin-top: 10px;
}

.listItem {
  flex-basis: 600px;
}
