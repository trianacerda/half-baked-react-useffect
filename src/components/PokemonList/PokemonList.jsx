import PropTypes from 'prop-types'
import Pokemon from '../Pokemon/Pokemon'
import './PokemonList.css'

export default function PokemonList({ pokemons }) {
  return (
    <ul aria-label="pokemonList" className="pokemonList">
      {pokemons.map((pokemon) => {
        return (
          <li className="listItem" key={pokemon.id}>
            <Pokemon pokemon={pokemon} />
          </li>
        )
      })}
    </ul>
  )
}

PokemonList.propTypes = {
  pokemons: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      abilityOne: PropTypes.string.isRequired,
      abilityTwo: PropTypes.string.isRequired,
      typeOne: PropTypes.string.isRequired,
      typeTwo: PropTypes.string.isRequired,
      pic: PropTypes.string.isRequired,
      generation: PropTypes.string.isRequired,
    })
  ).isRequired,
}
