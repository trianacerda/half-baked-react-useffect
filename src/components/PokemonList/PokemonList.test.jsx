import { render, screen } from '@testing-library/react'
import PokemonList from './PokemonList'

it('displays a list of pokemon', () => {
  render(
    <PokemonList
      pokemons={[
        {
          id: 7,
          name: 'Charizard',
          abilityOne: 'blaze',
          abilityTwo: 'NA',
          typeOne: 'fire',
          typeTwo: 'flying',
          pic: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/006.png',
          generation: 'First Generation',
        },
        {
          id: 172,
          name: 'Typhlosion',
          abilityOne: 'blaze',
          abilityTwo: 'NA',
          typeOne: 'fire',
          typeTwo: 'NA',
          pic: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/157.png',
          generation: 'Second Generation',
        },
        {
          id: 279,
          name: 'Blaziken',
          abilityOne: 'blaze',
          abilityTwo: 'NA',
          typeOne: 'fire',
          typeTwo: 'fighting',
          pic: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/257.png',
          generation: 'Third Generation',
        },
      ]}
    />
  )

  const pokemons = screen.getByRole('list', { name: 'pokemonList' })
  expect(pokemons).toMatchSnapshot()
})
