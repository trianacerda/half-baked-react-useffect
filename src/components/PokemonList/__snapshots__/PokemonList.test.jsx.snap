// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of pokemon 1`] = `
<ul
  aria-label="pokemonList"
  class="pokemonList"
>
  <li
    class="listItem"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="Charizard-First Generation"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/006.png"
      />
      <div
        class="details"
      >
        <h2>
          Charizard
        </h2>
        <p>
          First Generation
        </p>
        <p>
          blaze
        </p>
        <p>
          fire
          /flying
        </p>
      </div>
    </figure>
  </li>
  <li
    class="listItem"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="Typhlosion-Second Generation"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/157.png"
      />
      <div
        class="details"
      >
        <h2>
          Typhlosion
        </h2>
        <p>
          Second Generation
        </p>
        <p>
          blaze
        </p>
        <p>
          fire
        </p>
      </div>
    </figure>
  </li>
  <li
    class="listItem"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="Blaziken-Third Generation"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/257.png"
      />
      <div
        class="details"
      >
        <h2>
          Blaziken
        </h2>
        <p>
          Third Generation
        </p>
        <p>
          blaze
        </p>
        <p>
          fire
          /fighting
        </p>
      </div>
    </figure>
  </li>
</ul>
`;
